debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/PrintObjectFieldsHelper.java

```diff
@@ -196,7 +205,9 @@ public static void printJ9ObjectFields(PrintStream out, int tabLevel, J9ClassPoi
 							/* if nestedClassHierarchy is null then this class is the container class */
 							containerClazz = localClazz;
 						} else {
-							containerClazz = nestedClassHierarchy[0];
+							int containerClassIndex = Pattern.matches("\\[([1-9]\\d*|0)\\]", nestingHierarchy[0]) ? 1 : 0;
```

Comment:
I see no reason to treat `000` different than `0`; I think the pattern should be simplified to `[\d+]`.